src/integer/mat_z/set.rs

```diff
@@ -62,6 +78,118 @@ implement_for_owned!(Z, MatZ, SetEntry);
 
 implement_for_others!(Z, MatZ, SetEntry for i8 i16 i32 i64 u8 u16 u32 u64);
 
+impl MatrixSwaps for MatZ {
```

Comment:
swap to me implies swaps with extern matrices, I would prefer naming like `arrange, rearrange`, but swap is probably also fine. But iI would use the singular for the trait name to stay consistent

Reply:
I removed the trait now that it is not needed anymore for sorting. If needed anytime, we can always re-introduce it.
I sticked to the word "swap" as I believe that it is the more intuitive word to search for than "rearrange" or "arrange". You're correct that this should probably include also swapping elements with other matrices, but it would add at least one parameter to every function, which does not make the API any nicer.
Hence, I would stick to `get_column` and `set_column` (which will be implemented now) for inter-matrix swaps for now.